WorkAndFun/src/main/java/pl/infoshare/workandfun/announcements/AnnouncementController.java

```diff
@@ -23,17 +20,33 @@ public AnnouncementController(AnnouncementsService announcementsService) {
 
     @GetMapping("add-announcement")
     public String announcementForm(Model model){
-        model.addAttribute("announcement", new Announcement());
+        model.addAttribute("announcement", new AddAndEditDto());
         return "announcement-form";
     }
 
     @PostMapping("add-announcement")
-    public String save(@Valid @ModelAttribute("announcement") @RequestBody Announcement announcement,
+    public String save(@Valid @ModelAttribute("announcement") @RequestBody AddAndEditDto addAndEditDto,
                        BindingResult bindingResult) {
         if(bindingResult.hasErrors()){
             return "announcement-form";
         }
-        announcementsService.save(announcement);
+        announcementsService.save(addAndEditDto);
         return "announcement-form-success";
     }
+
+    @GetMapping("edit/{id}")
+    public String announcementEditForm(Model model, @PathVariable Long id){
+        model.addAttribute("announcement", announcementsService.findByIdConvertToDto(id));
+        return "announcement-form-update";
+    }
+
+    @PostMapping("edit/{id}")
```

Comment:
```suggestion
    @PutMapping("edit/{id}")
```
Zmieńmy na put + wpis do properties-ów + drobna zmiana w announcement-form-update